src/Microsoft.OpenApi/Writers/OpenApiWriterExtensions.cs

```diff
@@ -458,6 +459,17 @@ private static void WriteMapInternal<T>(
 
             if (elements != null)
             {
+                var settings = writer.GetSettings();
+
+                if (settings?.EnableSorting == true)
+                {
+                    elements = elements.Sort(); // sort with default comparer
+                }
+                else if (settings?.KeyComparer != null)
+                {
+                    elements = elements.Sort(settings.KeyComparer); // sort using custom comparer
```

Comment:
probably needs to be duplicated for lists/arrays as well

Reply:
This includes HashSets yes?

Reply:
I'd say yes according to Mike's comment in the original issue.